test/dynamo/test_misc.py

```diff
@@ -427,6 +427,33 @@ def h(x):
             cleanup_op("mylib::bar3")
             del lib
 
+    def test_auto_functionalize_can_with_default(self):
+        lib = torch.library.Library("mylib", "FRAGMENT")
+        torch.library.define(
+            "mylib::foo",
+            "(Tensor a, int b, Tensor(d!)? c=None, Tensor? d=None, int e=-1) -> ()",
+            tags=torch.Tag.pt2_compliant_tag,
+            lib=lib,
+        )
+
+        @torch.library.impl("mylib::foo", "cpu", lib=lib)
+        def foo_impl(a, b, c=None, d=None, e=-1):
+            a + mode
```

Comment:
Is this supposed to fail to compile?

Reply:
Rebasing mistake, waiting your change to force run the tests before fixing it.